apps/omg_api/lib/omg_api/block_queue.ex

```diff
@@ -147,17 +152,20 @@ defmodule OMG.API.BlockQueue do
         end
 
       submit_blocks(state1)
-      {:noreply, state1}
+      {:noreply, %Core{} = state1}
     end
 
-    def handle_cast({:enqueue_block, %Block{number: block_number, hash: block_hash} = block}, %Core{} = state) do
+    def handle_info(
+          {:internal_event_bus, :enqueue_block, %Block{number: block_number, hash: block_hash} = block},
+          %Core{} = state
+        ) do
       {:ok, parent_height} = Eth.get_ethereum_height()
       state1 = Core.enqueue_block(state, block_hash, block_number, parent_height)
       _ = Logger.info("Enqueuing block num '#{inspect(block_number)}', hash '#{inspect(Base.encode16(block_hash))}'")
 
       FreshBlocks.push(block)
```

Comment:
FreshBlocks push is handled in a separate PR right?

Reply:
no need to handle - both are on `OMG.API` so no separation of concerns concerns here. It makes some sense for `FreshBlocks` to get the block from the `BlockQueue`, so I decided to keep it simple. But you find it very unelegant or surprising, we can subscribe `FreshBlocks` too I think